lib/document.js

```diff
@@ -2003,7 +2003,8 @@ Document.prototype.$__dirty = function() {
   // gh-2558: if we had to set a default and the value is not undefined,
   // we have to save as well
   all = all.concat(this.$__.activePaths.map('default', function(path) {
-    if (path === '_id' || !_this.getValue(path)) {
+    var pType = typeof _this.getValue(path);
+    if (path === '_id' || (!_this.getValue(path) && pType !== 'boolean')) {
```

Comment:
This check is a little odd, and will still mess up with empty strings `''` because empty string is falsy. Can we change this to `|| _this.getValue(path) == null`?

Reply:
Odd indeed. fixed and test added for empty strings just in case I try to break it again 😀